File: csvToJson.py
# -*- coding: utf-8 -*-
"""
Created on Wed Oct 24 17:56:07 2018

@author: PShirbhate
"""

import os
os.chdir("D:/Users/PShirbhate/Downloads/Machine_Learning_A-Z-master/Machine_Learning_A-Z-master/Part 2 - Regression/Section 4 - Simple Linear Regression/")

import pandas as pd
dataset = pd.read_csv("Salary_Data.csv")

'convert CSV to JSON'

jsonobj = dataset.to_json(orient='split')
' other values for orient parameter - index, records, table'

'read .json file and convert it to dataframe'

import json

with open('sampledata.json') as json_data:
    d = json.load(json_data)
    print(d)
    
dataset1 = pd.DataFrame(d)

from pandas.io.json import json_normalize
print json_normalize(d)

File: csvToJson.R

setwd("D:/Users/PShirbhate/Downloads/Machine_Learning_A-Z-master/Machine_Learning_A-Z-master/Part 2 - Regression/Section 4 - Simple Linear Regression/")
  
dataset = read.csv("Salary_Data.csv")

#install.packages("rjson")

library(rjson)

# convert csv to json
dfjson = toJSON(dataset)

cat(dfjson)

str(dfjson)
# save json object as json file 
write(dfjson, file="sampledata.JSON")

# read .json file and convert it to dataframe
json_data <- fromJSON(file="sampledata.JSON")



# replacing null with NA's
dffromJson <- lapply(json_data, function(x) {
  x[sapply(x, is.null)] <- NA
  unlist(x)
})

dffromJson1 <- do.call("rbind",json_data)
dffromJson2 <- as.data.frame(dffromJson1)
str(dffromJson2)

dffromJson1 <- do.call("cbind",json_data)
dffromJson2 <- as.data.frame(dffromJson1)
